src/main/kotlin/com/netflix/dgs/plugin/hints/DgsInputArgumentInspector.kt

```diff
@@ -0,0 +1,237 @@
+/*
+ * Copyright 2021 Netflix, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.netflix.dgs.plugin.hints
+
+import com.intellij.codeInspection.*
+import com.intellij.lang.jsgraphql.psi.*
+import com.intellij.lang.jsgraphql.psi.impl.GraphQLFieldDefinitionImpl
+import com.intellij.lang.jsgraphql.psi.impl.GraphQLIdentifierImpl
+import com.intellij.lang.jsgraphql.psi.impl.GraphQLTypeNameImpl
+import com.intellij.openapi.project.Project
+import com.intellij.psi.*
+import com.intellij.psi.util.parentOfType
+import com.intellij.uast.UastVisitorAdapter
+import com.netflix.dgs.plugin.MyBundle
+import com.netflix.dgs.plugin.services.DgsService
+import org.jetbrains.kotlin.psi.KtFile
+import org.jetbrains.kotlin.psi.KtFunction
+import org.jetbrains.kotlin.psi.KtPsiFactory
+import org.jetbrains.kotlin.util.removeSuffixIfPresent
+import org.jetbrains.uast.*
+import org.jetbrains.uast.visitor.AbstractUastNonRecursiveVisitor
+import java.time.*
+import java.util.*
+import kotlin.system.measureTimeMillis
+
+class DgsInputArgumentInspector : AbstractBaseUastLocalInspectionTool() {
+    override fun buildVisitor(holder: ProblemsHolder, isOnTheFly: Boolean): PsiElementVisitor {
+
+        return UastVisitorAdapter(object : AbstractUastNonRecursiveVisitor() {
+            override fun visitMethod(node: UMethod): Boolean {
+                val time = measureTimeMillis {
+
+                    if (node.hasAnnotation(DGS_DATA_ANNOTATION) || node.hasAnnotation((DGS_QUERY_ANNOTATION)) ) {
+                        val dgsDataAnnotation = if (node.getAnnotation(DGS_DATA_ANNOTATION) != null) { node.getAnnotation(DGS_DATA_ANNOTATION) } else { node.getAnnotation(DGS_QUERY_ANNOTATION) }
+                        val dgsService = dgsDataAnnotation!!.project!!.getService(DgsService::class.java)
+                        val dgsDataFetcher = dgsService.dgsComponentIndex.dataFetchers.find { it.psiAnnotation.toUElement() == dgsDataAnnotation.toUElement() }
+                        val isJavaFile = dgsDataFetcher?.psiFile is PsiJavaFile
```

Comment:
I don’t think this works for Kotlin

Reply:
![Screen Shot 2021-12-06 at 2 25 37 PM](https://user-images.githubusercontent.com/41588701/144932649-b9024648-1a7e-4ba6-aa12-4df289fcd016.png)

The psiFile is a ktFile in case of kotlin, so the check work fine from what I've seen